nav2-usability/scenario-code/deeplink-pathparam/router.dart

```diff
@@ -18,6 +24,47 @@ class Book {
   Book(this.title, this.author);
 }
 
+class AppState extends ChangeNotifier {
+  int? _selectedBookId;
+
+  int? get selectedBookId => _selectedBookId;
+
+  set selectedBookId(int? id) {
+    _selectedBookId = id;
+    notifyListeners();
+  }
+
+  Book? get selectedBook {
```

Comment:
App state should not be in charge of looking up which books. Since the books has been moved out of the app state, it should be a global function, or just `books[_appState.selectedBookId]`

Reply:
The idea is that these books are part of the app state (e.g. HTTP request) - maybe I should move `books` back to AppState?

Reply:
If the app meant to cache the books from the http request, then yes.

Reply:
okay, I think that makes the most sense here. LMKWYT

Reply:
sgtm